extensions/amp-social-share/0.2/amp-social-share.js

```diff
@@ -0,0 +1,178 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {Services} from '../../../src/services';
+import {SocialShare} from './social-share';
+import {addParamsToUrl} from '../../../src/url';
+import {dev, user, userAssert} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {getDataParamsFromAttributes} from '../../../src/dom';
+import {getSocialConfig} from './amp-social-share-config';
+import {isExperimentOn} from '../../../src/experiments';
+import {toggle} from '../../../src/style';
+
+/** @const {string} */
+const TAG = 'amp-social-share';
+
+class AmpSocialShare extends PreactBaseElement {
+  /** @override */
+  init() {
+    const viewer = Services.viewerForDoc(this.element);
+    const platform = Services.platformFor(window);
+    const typeConfig = this.getTypeConfigOrUndefined_(
+      this.element.getAttribute('type'),
+      viewer,
+      platform
+    );
+    // Hide/ignore component if typeConfig is undefined
+    if (!typeConfig) {
+      toggle(this.element, false);
+      return;
+    }
+    this.setHrefAndTargetContext_(typeConfig, platform);
+  }
+
+  /** @override */
+  isLayoutSupported() {
+    userAssert(
+      isExperimentOn(this.win, 'amp-social-share-v2'),
+      'expected amp-social-share-v2 experiment to be enabled'
+    );
+    return true;
+  }
+
+  /**
+   * @private
+   * @param {?string} type
+   * @param {!../../../src/service/viewer-interface.ViewerInterface} viewer
+   * @param {!../../../src/service/platform-impl.Platform} platform
+   * @return {!JsonObject|undefined}
+   */
+  getTypeConfigOrUndefined_(type, viewer, platform) {
+    userAssert(type, 'The type attribute is required. %s', this.element);
+    userAssert(
+      !/\s/.test(type),
+      'Space characters are not allowed in type attribute value. %s',
+      this.element
+    );
+    if (type === 'system') {
+      // navigator.share unavailable
+      if (!this.systemShareSupported_(viewer, platform)) {
+        return;
+      }
+    } else {
+      // system share wants to be unique
+      const systemOnly =
+        this.systemShareSupported_(viewer, platform) &&
+        !!window.document.querySelector(
+          'amp-social-share[type=system][data-mode=replace]'
+        );
+      if (systemOnly) {
+        return;
+      }
+    }
+    const typeConfig = getSocialConfig(type) || dict();
```

Comment:
If we can't find a type config, shouldn't we be returning nothing? As is, this returns an empty dict, which will let the element build.

Reply:
We do want it to still build since people may be using non-configured providers: https://amp.dev/documentation/components/amp-social-share/#non-configured-providers